lib/ObjWriter/objwriter.cpp

```diff
@@ -559,22 +556,119 @@ extern "C" void EmitDebugFileInfo(ObjectWriter *OW, int FileId,
   }
 }
 
-// This should be invoked at the end of function code emission to flush
-// debug function info.
-extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
-                                      const char *FunctionName,
-                                      int FunctionSize) {
+static void EmitSymRecord(MCObjectStreamer &OST, int Size,
+                          SymbolRecordKind SymbolKind) {
+  SymRecord Rec = {ulittle16_t(Size + sizeof(ulittle16_t)),
+                   ulittle16_t(SymbolKind)};
+  OST.EmitBytes(StringRef((char *)&Rec, sizeof(Rec)));
+}
 
+static void EmitVarDefRange(MCObjectStreamer &OST, const MCSymbol *Fn,
+                            LocalVariableAddrRange &Range) {
+  const MCSymbolRefExpr *BaseSym =
+      MCSymbolRefExpr::create(Fn, OST.getContext());
+  const MCExpr *Offset =
+      MCConstantExpr::create(Range.OffsetStart, OST.getContext());
+  const MCExpr *Expr =
+      MCBinaryExpr::createAdd(BaseSym, Offset, OST.getContext());
+  OST.EmitCOFFSecRel32Value(Expr);
+  OST.EmitCOFFSectionIndex(Fn);
+  OST.EmitIntValue(Range.Range, 2);
+}
+
+static void EmitCVDebugVarInfo(MCObjectStreamer &OST, const MCSymbol *Fn,
+                               DebugVarInfo LocInfos[], int NumVarInfos) {
+  for (int I = 0; I < NumVarInfos; I++) {
+    // Emit an S_LOCAL record
+    DebugVarInfo Var = LocInfos[I];
+    LocalSym Sym = {};
+
+    int SizeofSym = sizeof(LocalSym);
+    EmitSymRecord(OST, SizeofSym + Var.Name.length() + 1,
+                  SymbolRecordKind::S_LOCAL);
+
+    TypeIndex TypeIndex(Var.TypeIndex);
+    Sym.Type = TypeIndex;
```

Comment:
Nit: Using the same type/local name,  TypeIndex is odd. Can you just inline the above to the below.